### ISSUE TITLE
Absolute path to package is presented in the build output assets

### ISSUE THREAD

**Issue by stevcooo** [embroider-build/embroider#1243]
**Describe the bug**
When some package is using [embroider-build/embroider](https://github.com/embroider-build/embroider/releases/tag/v1.8.3), it includes the absolute path in the output assets. We discovered this by using the [ember-fontawesome](https://github.com/FortAwesome/ember-fontawesome) and we reported an [Issue there](https://github.com/FortAwesome/ember-fontawesome/issues/212) and they redirected us to [ember-get-config](https://github.com/mansona/ember-get-config) and we opened [an issue there](https://github.com/mansona/ember-get-config/issues/51) too but they redirected us to you.

**Reproducible test case**
Here I've created a repo where the issue can be reproduced easily [fa-test](https://github.com/stevcooo/fa-test).  If you want you can directly examine the [vendor file](https://github.com/stevcooo/fa-test/blob/master/dist/assets/vendor-1d34ecfc5e3d346ecac926c16af3ad84.js#L5031).   
Or get the repo, and do  `npm install` and examine `vendor...js` file in `dist` folder, you should see an absolute path to one of the packages in your local machine.

**Expected behavior**
We want deterministic builds that are absolute-path invariant (i.e. the build output shouldn't depend on the machine or the project's location in the file system). Sometimes we build on our local machine, but sometimes the build is done in some agent (CI/CD Azure for example). We want build outputs to be identical no matter the machine they are built on.

**Desktop (please complete the following information):**
 - Browser: Chrome
 - Version: 104.0.5112.81 (Official Build) (64-bit)

**Additional context**
Please check the [ReadMe](https://github.com/stevcooo/fa-test#readme) file for more instructions about this issue.

**Comment by ef4** [comment]
Thanks for reporting.

I think the bug here is that we're leaving runtime information in the build even in production builds that resolve all macros at build-time. I'm pretty sure this bit of code that includes the absolute paths isn't getting used at all anyway. 

The macros work differently in dev vs prod. In dev builds we do more work at runtime, which makes it much cheaper to switch between environments (like testing vs not-testing and browser vs fastboot). In prod builds we inline all the macro results at build time, which results in smaller, faster output at the cost of needing to produce entirely separate assets for different environments (primarily browser vs fastboot).

This bug probably only affects people on classic builds. Under embroider, this entire module gets optimized away because nothing is using it.

I think this might be the fix but I haven't had a chance to test it:

```diff
diff --git a/packages/macros/src/babel/macros-babel-plugin.ts b/packages/macros/src/babel/macros-babel-plugin.ts
index 342f4e5b..385f2380 100644
--- a/packages/macros/src/babel/macros-babel-plugin.ts
+++ b/packages/macros/src/babel/macros-babel-plugin.ts
@@ -47,7 +47,7 @@ export default function main(context: typeof Babel): unknown {
     FunctionDeclaration: {
       enter(path: NodePath<t.FunctionDeclaration>, state: State) {
         let id = path.get('id');
-        if (id.isIdentifier() && id.node.name === 'initializeRuntimeMacrosConfig') {
+        if (id.isIdentifier() && id.node.name === 'initializeRuntimeMacrosConfig' && state.opts.mode === 'run-time') {
           let pkg = state.owningPackage();
           if (pkg && pkg.name === '@embroider/macros') {
             inlineRuntimeConfig(path, state, context);
```

**Comment by stevcooo** [comment]
Thanks for the fast replay, It works!
I'll prepare a PR soon and I will link it to this issue. Or if you have any other approaches to situations like this one please let me know.

**Comment by stevcooo** [comment]
Thanks for the fast replay, It works!
I've prepared this [PR](https://github.com/embroider-build/embroider/pull/1245)  soon and I will link it to this issue. Or if you have any other approaches to situations like this one please let me know.

**Comment by stevcooo** [comment]
Hi @ef4, sorry for bothering you, I've submited this PR previous week. Some steps of the build were failing, today I've rebased it, but to run the workflows in my [PR](https://github.com/embroider-build/embroider/pull/1245), they need to be approved by some maintainer. Could you please do that, or should I contact somebody else?

**Comment by stevcooo** [comment]
Hi @ef4 , me again, can you please guide me what should I do next, please take a look at this video when you can:
https://www.loom.com/share/34bd002945dd413bba0bd1ce3bfbc0bc

**Comment by ef4** [comment]
Fixed in #1245
